config/initializers/new_framework_defaults_7_1.rb

```diff
@@ -0,0 +1,87 @@
+# Be sure to restart your server when you modify this file.
+#
+# This file eases your Rails 7.1 framework defaults upgrade.
+#
+# Uncomment each configuration one by one to switch to the new default.
+# Once your application is ready to run with all new defaults, you can remove
+# this file and set the `config.load_defaults` to `7.1`.
+#
+# Read the Guide for Upgrading Ruby on Rails for more info on each option.
+# https://guides.rubyonrails.org/upgrading_ruby_on_rails.html
+
+# 💥✅ I've removed the lines in this file where I agree with the new defaults
+# or when they are not applicable to us. I've kept the lines that I'm not sure about
+# or that I think we should discuss.
+# TODO: We still have to activate the new defaults by setting
+# `config.load_defaults` to `7.1` (!)
+
+# ❓ Parameters Hash -- Not sure whether this affects us
+###
+# Do not treat an `ActionController::Parameters` instance
+# as equal to an equivalent `Hash` by default.
+#++
+# Rails.application.config.action_controller.allow_deprecated_parameters_hash_equality = false
+
+# ❓ Invalid cache expiration time, what behavior do we want?
+###
+# Specify if an `ArgumentError` should be raised if `Rails.cache` `fetch` or
+# `write` are given an invalid `expires_at` or `expires_in` time.
+# Options are `true`, and `false`. If `false`, the exception will be reported
+# as `handled` and logged instead.
+#++
+# Rails.application.config.active_support.raise_on_invalid_cache_expiration_time = true
+
+# ❓ New default serializer
+###
+# Specify the default serializer used by `MessageEncryptor` and `MessageVerifier`
+# instances.
+#
+# The legacy default is `:marshal`, which is a potential vector for
+# deserialization attacks in cases where a message signing secret has been
+# leaked.
+#
+# In Rails 7.1, the new default is `:json_allow_marshal` which serializes and
+# deserializes with `ActiveSupport::JSON`, but can fall back to deserializing
+# with `Marshal` so that legacy messages can still be read.
+#
+# In Rails 7.2, the default will become `:json` which serializes and
+# deserializes with `ActiveSupport::JSON` only.
+#
+# Alternatively, you can choose `:message_pack` or `:message_pack_allow_marshal`,
+# which serialize with `ActiveSupport::MessagePack`. `ActiveSupport::MessagePack`
+# can roundtrip some Ruby types that are not supported by JSON, and may provide
+# improved performance, but it requires the `msgpack` gem.
+#
+# For more information, see
+# https://guides.rubyonrails.org/v7.1/configuring.html#config-active-support-message-serializer
+#
+# If you are performing a rolling deploy of a Rails 7.1 upgrade, wherein servers
+# that have not yet been upgraded must be able to read messages from upgraded
+# servers, first deploy without changing the serializer, then set the serializer
+# in a subsequent deploy.
+#++
+# Rails.application.config.active_support.message_serializer = :json_allow_marshal
```

Comment:
I'm not even sure if [this](https://guides.rubyonrails.org/v7.1/configuring.html#config-active-support-message-serializer) affects us. Do we make use of the `ActiveSupport:MessageEncryptor` or the `ActiveSupport::MessageVerifier`?

If no, maybe we should use the default for Rails 7.2, which will be `:json`? Otherwise, we could also try `:message_pack`, but that requries the `msgpack` gem, which we don't have in our Gemfile right now. If we're not using this anyways, `:json` is a viable option I guess.

Reply:
We do not use this (at least not actively), so `:json` seems fine to me.